File: src/main/java/com/amvholdings/reports/component/invoice/InvoiceModel.java
package com.amvholdings.reports.component.invoice;

import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Getter
@Setter
public class InvoiceModel {
    Long id;
    String sourceSystem;
    String sourceId;
    String invoiceId;
    String paymentDate;
    String createdAt;
    String updatedAt;
    BigDecimal paymentAmount;
    BigDecimal grandTotal;
    BigDecimal shippingAmount;
    BigDecimal taxAmount;
    InvoiceAddressModel shippingAddress = new InvoiceAddressModel();
    InvoiceAddressModel billingAddress = new InvoiceAddressModel();
}

File: src/app/sales/invoices/shared/invoice.model.ts
import {Address} from "./address.model";

export class Invoice {
    id: number;
    sourceSystem: string;
    sourceId: string;
    invoiceId: string;
    paymentDate: string;
    createdAt: string;
    updatedAt: string;
    paymentAmount: number;
    grandTotal: number;
    shippingAmount: number;
    taxAmount: number;
    register: string;
    customerId: number;
    shippingAddress: Address;
    billingAddress: Address;
    customerAddresses: Address[];
}
